src/components/MagicCodeInput.js

```diff
@@ -115,7 +115,21 @@ class MagicCodeInput extends React.PureComponent {
      * @param {Object} event
      * @param {Number} index
      */
-    onFocus(event, index) {
+    onFocus(event) {
+        event.preventDefault();
+        this.setState({
+            input: '',
+        });
+    }
```

Comment:
Not setting the focused index onFocus caused a regression https://github.com/Expensify/App/issues/23596